### PR TITLE
Fix missing equipment for mecha

### PR DESCRIPTION
**Pull request by kreeperHLC**
## About The Pull Request
Fixing the disappearance of equipment when trying to install it in the wrong type of mecha.

## Changelog
:cl:
fix: Equipment no longer disappears when you try to install it in the wrong type of mecha.
/:cl:

### CODE REVIEW

**Comment by francinum** [comment]
Check if the item can even be attached first before taking it in the first place?

**Comment by kreeperHLC** [comment]
> Check if the item can even be attached first before taking it in the first place?

I understand that you want me to check the attachment of the equipment when it is not in your hand, but how do you imagine that? Neither I nor my head could think of a way to do it.
/obj/item/mecha_parts/mecha_equipment/try_attach_part exists only in /obj/mecha/attackby

**Comment by kreeperHLC** [comment]
I decided to fix it in a simpler way. I switched the checks around. Now you don't even have to wait for an equipment to be attached to see if it doesn't match.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Jeez to be after do_mob

**Comment by francinum** [comment]
> 
> 
> Jeez to be after do_mob

Clarify?

**Comment by PowerfulBacon** [comment]
The check needs to be after the do_after check due to that having a delay.

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 1]
```diff
@@ -48,13 +48,13 @@
 	return ..()
 
 /obj/item/mecha_parts/mecha_equipment/try_attach_part(mob/user, obj/mecha/M)
+	if(!can_attach(M))
+		to_chat(user, "<span class='warning'>You are unable to attach [src] to [M]!</span>")
+		return FALSE
 	if(!do_mob(user, M, 15))
 		return FALSE
 	if(!user.temporarilyRemoveItemFromInventory(src))
```
Comment (ike709) [inline code/game/mecha/equipment/mecha_equipment.dm:56]: The check should be between do_mob() and this

**Comment by kreeperHLC** [comment]
I don't know why you make people wait, but like you said, that's what I did. /shrug

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**